tests/unit/maze_transformer/training/test_tokenizer.py

```diff
@@ -0,0 +1,52 @@
+import re
+
+import numpy as np
+
+from maze_transformer.generation.generators import LatticeMazeGenerators
+from maze_transformer.training.mazedataset import MazeDatasetConfig
+from maze_transformer.training.tokenizer import SPECIAL_TOKENS, MazeTokenizer
+from tests.helpers import utils
+
+
+def test_coordinate_system():
+    """
+    Check that the adjlist created by .as_tokens() uses the same coordinate system as the LatticeMaze adjlist.
+
+    To test this, generate both adjlists, sort them and convert to a common format, and check that they are equal.
+    """
+    maze_size = 3
+    maze = LatticeMazeGenerators.gen_dfs((maze_size, maze_size))
+    maze_adjlist = maze.as_adj_list()
+
+    # convert to the same format as the tokenizer adjlist
+    maze_adjlist_connections = [
+        [f"({c[0]},{c[1]})" for c in conn] for conn in maze_adjlist
+    ]
+
+    # See https://github.com/AISC-understanding-search/maze-transformer/issues/77
+    node_token_map = MazeDatasetConfig(
+        grid_n=maze_size, name="test", n_mazes=1
+    ).node_token_map
+
+    tokenized_maze = MazeTokenizer(
+        maze=maze,
+        solution=np.array([[0, 0]]),
+    ).as_tokens(node_token_map)
+
+    tokenizer_adjlist = tokenized_maze[
+        tokenized_maze.index(SPECIAL_TOKENS["adjlist_start"])
+        + 1 : tokenized_maze.index(SPECIAL_TOKENS["adjlist_end"])
+    ]
+
+    # remove special tokens
+    tokenizer_adjlist_coordinates = [
+        token for token in tokenizer_adjlist if re.search(r"\(\d+,\d+\)", token)
```

Comment:
Just messing around:

```python
    
    ...

    tokenizer_adjlist = tokenized_maze[
        tokenized_maze.index(SPECIAL_TOKENS["adjlist_start"])
        + 1 : tokenized_maze.index(SPECIAL_TOKENS["adjlist_end"])
    ]

    tokenizer_adjlist_connections = pairs(
        i for i in tokenizer_adjlist if i not in SPECIAL_TOKENS.values()
    )

    assert utils.adjlist_to_nested_set(
        tokenizer_adjlist_connections
    ) == utils.adjlist_to_nested_set(maze_adjlist_connections)

def pairs(t):
    it = iter(t)
    return list(zip(it, it))
```

had a bit of help from https://stackoverflow.com/questions/4628290/pairs-from-single-list

Reply:
Great call on `if token not in SPECIAL_TOKENS`! I'll do that.

I also saw that SO thread, but decided against the `iter` approach because I found it less intuitive than just using a `zip`. Might just be my lack of experience with Python though.